src/main/python/main.py

```diff
@@ -0,0 +1,1123 @@
+        obj.setLocation(PhysicalLocation(uri, Region(node.lineno,
+                                                     node.col_offset, node.end_lineno, node.end_col_offset)))
+        obj.setCode(self.sourcecode.get_snippet(node.lineno,
+                                                node.col_offset, node.end_lineno, node.end_col_offset))
+        # obj.setCode(ast.unparse(node)) # alternative to CodeExtractor class
+
+    ### LITERALS ###
+    def visit_Constant(self, node):
+        debug_print(ast.dump(node))
+        lit = Literal()
+        self.add_loc_info(node, lit)
+        lit.setValue(node.value)
+        lit.setName(str(node.value))
+        if type(node.value) is str:
+            lit.setType(TypeParser.createFrom("str", False))
+        elif type(node.value) is int:
+            lit.setType(TypeParser.createFrom("int", False))
+        elif type(node.value) is float:
+            lit.setType(TypeParser.createFrom("float", False))
+        elif type(node.value) is complex:
+            lit.setType(TypeParser.createFrom("complex", False))
+        elif type(node.value) is bool:
+            lit.setType(TypeParser.createFrom("bool", False))
+        elif type(node.value) is type(None):
+            lit.setType(TypeParser.createFrom("None", False))
+        elif type(node.value) is bytes:
+            lit.setType(TypeParser.createFrom("byte[]", False))
+        else:
+            debug_print(type(node.value))
+            raise NotImplementedError
+        return lit
+
+    def visit_FormattedValue(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    def visit_JoinedStr(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    def visit_List(self, node):
+        debug_print(ast.dump(node))
+
+        ile = InitializerListExpression()
+        self.add_loc_info(node, ile)
+
+        list = ArrayList()
+
+        for el in node.elts:
+            expr = self.visit(el)
+            list.add(expr)
+
+        ile.setInitializers(list)
+
+        return ile
+
+    def visit_Tuple(self, node):
+        # debug_print(ast.dump(node))
+        # lit = Literal()
+        # self.add_loc_info(node, lit)
+        # lit.setType(TypeParser.createFrom("Tuple", False))
+        # elts = []
+        # for e in node.elts:
+        #    elts.append(self.visit(e))
+        # lit.setValue(elts)
+        return Expression()
+
+    def visit_Set(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    def visit_Dict(self, node):
+        debug_print(ast.dump(node))
+        # TODO implement it
+        # raise NotImplementedError
+        return Expression()
+
+    ### VARIABLES ###
+    def visit_Name(self, node):
+        debug_print(ast.dump(node))
+        ref = DeclaredReferenceExpression()
+        self.add_loc_info(node, ref)
+        ref.setName(node.id)
+
+        # resolved_ref = self.scopemanager.resolve(ref)
```

Comment:
It *might* be necessary to call `ScopeManager::connectToLocal` with the `ref`. We do this for Java and Golang, but not for C++ for some reason. This is unfortunately very inconsistent, but @konradweiss is aware of this.